From 13808b9347bf4c96b5d3a32201e599687e2c9fba Mon Sep 17 00:00:00 2001
From: Kurt Kiefer <kurt.kiefer@arthrex.com>
Date: Mon, 15 Aug 2022 14:09:40 -0700
Subject: [PATCH] Fix building extensions with complex CXX

Build tools like Yocto qualify the compiler command with the sysroot
path and other arguments, this change handlers that during compiler
detection.
---
 torch/utils/cpp_extension.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/torch/utils/cpp_extension.py b/torch/utils/cpp_extension.py
index 00e6d5d4..31a2b052 100644
--- a/torch/utils/cpp_extension.py
+++ b/torch/utils/cpp_extension.py
@@ -254,6 +254,7 @@ def check_compiler_ok_for_platform(compiler: str) -> bool:
     '''
     if IS_WINDOWS:
         return True
+    compiler = compiler.split()[0]
     which = subprocess.check_output(['which', compiler], stderr=subprocess.STDOUT)
     # Use os.path.realpath to resolve any symlinks, in particular from 'c++' to e.g. 'g++'.
     compiler_path = os.path.realpath(which.decode(*SUBPROCESS_DECODE_ARGS).strip())
